portal-impl/src/com/liferay/portal/service/impl/ResourcePermissionLocalServiceImpl.java

```diff
@@ -362,8 +362,11 @@ public ResourcePermission fetchResourcePermission(
 			resourcePermissions.size());
 
 		for (ResourcePermission resourcePermission : resourcePermissions) {
+			Set<String> emptySet = Collections.emptySet();
+
 			if (resourcePermission.getActionIds() == 0) {
-				continue;
+				roleIdsToActionIds.put(
```

Comment:
1) Oh yeah, I guess we should continue there. I don't think any of the following code needs to be executed.

2) I started out with that and got some sort of error because it didn't realize that Collections.emptySet() was of type Set<String>. This might've just been an IntelliJ warning though; I don't remember whether or not it actually stopped it from compiling.